Advertise correct MySQL server version (#12196)
gravitational/teleport: docs/pages/database-access/reference/configuration.mdx

    ...
    -----END CERTIFICATE-----

  # Additional database options.
  options:

>     # MySQL server version reported by Teleport Proxy in tunnel mode when the real version is not available.

Reviewer: Is there another way to phrase "the real version"? Why would the MySQL server version not be the real version?

(I haven't looked at this PR beyond this one comment, so apologies if the answer is elsewhere.)

Author: I've tried my best 😅

    ...
    -----END CERTIFICATE-----

  # MySQL only options.
  mysql:

>     # The default MySQL server version reported by Teleport Proxy.

Reviewer: ```suggestion
    # The default MySQL server version reported by the Teleport Proxy Service.
```

>     # The default MySQL server version reported by the Teleport Proxy Service.

Reviewer: ```suggestion
    # The MySQL server version reported by the Teleport Proxy Service.
    # Teleport uses this string when reporting the server version to a 
    # connecting client.
    #
    # When this option is not set, the Database Service will try to connect to
    # a MySQL instance on startup and fetch the server version. Otherwise, 
    # it will use the provided value without connecting to a database. 
    #
    # In both cases, the MySQL server version reported to a client will be 
    # updated on the first successful connection made by a user.
```

>     # When this option is set the Database Agent doesn't try to check the MySQL server version.

Reviewer: ```suggestion
    # When this option is set, the Database Service does not try to check the MySQL server version.
```

Reviewer: And the Database Service assumes that the server version is the one provided?

Author: Yes, this is a fallback version that will be used until we're able to get the version from the database. In most cases this value will never be used. If database is down during during Teleport startup then we will use this version for very first connection, as we were not able to fetch the version earlier.

>     # When this option is set, the Database Service does not try to check the MySQL server version.

Reviewer: I think this needs a sentence about how the Database Service uses the MySQL server version and when a user would use this option. Otherwise, the negative statement, "the Database Service does not try to check..." doesn't tell us much.

Also, does this field need to be present (but an empty string) if it's unused, or can the user leave it out?

Author: @ptgott How about:

```yaml
# The default MySQL server version reported by the Teleport Proxy Service.
# Teleport uses that string instead of default '8.0.0-Teleport' version when reporting 
# the server version to a connecting client. When this option is not set, the Database Service will try 
# to connect to MySQL instance on startup and fetch the server version. 
# Otherwise, it will use the provided value without connecting to a database.
# In both cases MySQL server version reported to a client will be updated on the first successful 
# connection made by a user.
```

> Also, does this field need to be present (but an empty string) if it's unused, or can the user leave it out?

When unused should not be in the configuration, but if someone decide to use empty string we are defaulting to `8.0.0-Teleport` - this is the same value as we're using right now.

>     # When this option is set, the Database Service does not try to check the MySQL server version.

Reviewer: ```suggestion
```
